optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/drools/uni/DroolsValuePair.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.drools.uni;
+
+import java.util.Objects;
+
+public final class DroolsValuePair<K,V> {
+    public final K key;
+    public final V value;
+    private final int hashCode;
+
+    public DroolsValuePair(K key, V value) {
+        this.key = key;
+        this.value = value;
+        this.hashCode = Objects.hash(key, value);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !Objects.equals(getClass(), o.getClass())) {
+            return false;
+        }
+        final DroolsValuePair<?, ?> pair = (DroolsValuePair<?, ?>) o;
+        return Objects.equals(key, pair.key) &&
+                Objects.equals(value, pair.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return hashCode;
+    }
+
+    @Override
+    public String toString() {
+        return "Pair[" + key + ", " + value + "]";
```

Comment:
In toString() and log messages,
we use "[" to denote lists, collections, arrays, etc.
We use "(" to denote fields.
So let's change this into: return "Pair(" + key + ", " + value + ")";